p2p/src/test/kotlin/maru/p2p/P2PTest.kt

```diff
@@ -585,6 +591,78 @@ class P2PTest {
     }
   }
 
+  @Test
+  fun `peer send a status request and receive exception when callee throws error`() {
+    val p2PNetworkImpl1 =
+      P2PNetworkImpl(
+        privateKeyBytes = key1,
+        p2pConfig =
+          P2P(
+            ipAddress = IPV4,
+            port = PORT1,
+            staticPeers = emptyList(),
+          ),
+        chainId = chainId,
+        serDe = RLPSerializers.SealedBeaconBlockSerializer,
+        metricsFacade = TestMetrics.TestMetricsFacade,
+        statusMessageFactory =
+          StatusMessageFactory(
+            beaconChain = beaconChain,
+            forkIdHashProvider = { throw IllegalStateException("currentForkIdHash exception testing") },
+          ),
+        beaconChain = beaconChain,
+        metricsSystem = TestMetrics.TestMetricsSystemAdapter,
+        forkIdHashProvider = forkIdHashProvider,
+        isBlockImportEnabledProvider = { true },
+      )
+    val p2pManagerImpl2 =
+      P2PNetworkImpl(
+        privateKeyBytes = key2,
+        p2pConfig =
+          P2P(
+            ipAddress = IPV4,
+            port = PORT2,
+            staticPeers = listOf(PEER_ADDRESS_NODE_1),
+          ),
+        chainId = chainId,
+        serDe = RLPSerializers.SealedBeaconBlockSerializer,
+        metricsFacade = TestMetrics.TestMetricsFacade,
+        statusMessageFactory =
+          StatusMessageFactory(
+            beaconChain = beaconChain,
+            forkIdHashProvider = { throw IllegalStateException("currentForkIdHash exception testing") },
+          ),
+        beaconChain = beaconChain,
+        metricsSystem = TestMetrics.TestMetricsSystemAdapter,
+        forkIdHashProvider = forkIdHashProvider,
+        isBlockImportEnabledProvider = { true },
+      )
+    try {
+      p2PNetworkImpl1.start()
+      p2pManagerImpl2.start()
+
+      awaitUntilAsserted { assertNetworkHasPeers(network = p2PNetworkImpl1, peers = 1) }
+      awaitUntilAsserted { assertNetworkHasPeers(network = p2pManagerImpl2, peers = 1) }
+
+      val peer1 =
+        p2pManagerImpl2.getPeerLookup().getPeer(LibP2PNodeId(PeerId.fromBase58(PEER_ID_NODE_1)))
+          ?: throw IllegalStateException("Peer with ID $PEER_ID_NODE_1 not found in p2pManagerImpl2")
+      val maruPeer1 =
+        DefaultMaruPeer(peer1, rpcMethods, statusMessageFactory, p2pConfig = P2P(ipAddress = IPV4, port = PORT1))
+
+      val responseFuture = maruPeer1.sendStatus()
+
+      assertThatThrownBy { responseFuture.get() }
```

Comment:
Since failing `forkIdHashProvider` is set on both sides, I don't know if this error comes from the caller or the callee here. I think it's better to pass some proper fork id provider to the `p2PNetworkImpl1`